File: 3172-divisible-and-non-divisible-sums-difference/divisible-and-non-divisible-sums-difference.cpp
class Solution {
public:
    int differenceOfSums(int n, int m) {
        int num1 = 0;
        int num2 = 0;

        // Build the num1 and num2
        for(int i = 1; i <= n; ++i) {
            if (i % m != 0) {
                num1 += i;
            } else {
                num2 += i;
            }
        }

        return num1 - num2;
    }

    /*
        Analysis:
            Time Complexity  : O(n)
            Space Complexity : O(1)
    */
};

File: 3172-divisible-and-non-divisible-sums-difference/divisible-and-non-divisible-sums-difference.java
class Solution {

    // Function used to solve the given problem
    private int solve(int n, int m) {
        int result = 0;

        for (int counter = 1; counter <= n; ++counter) {
            if (counter % m != 0) {
                result += counter;
            } else {
                result -= counter;
            }
        }

        return result;
    }
    /*
        Analysis:
            Time Complexity  : O(n)
            Space Complexity : O(1)
    */

    public int differenceOfSums(int n, int m) {
        return solve(n,m);
    }
}